coala_langserver/langserver.py

```diff
@@ -124,17 +146,11 @@ def main():
 
     if args.mode == 'stdio':
         log('Reading on stdin, writing on stdout')
-        s = LangServer(conn=ReadWriter(sys.stdin, sys.stdout))
-        s.listen()
+        start_io_lang_server(LangServer, sys.stdin.buffer, sys.stdout.buffer)
     elif args.mode == 'tcp':
         host, addr = '0.0.0.0', args.addr
         log('Accepting TCP connections on {}:{}'.format(host, addr))
-        ThreadingTCPServer.allow_reuse_address = True
-        s = ThreadingTCPServer((host, addr), LangserverTCPTransport)
-        try:
-            s.serve_forever()
-        finally:
-            s.shutdown()
+        start_tcp_lang_server(host, addr, LangServer)
```

Comment:
`host, addr, LangServer` ?

would have been a good idea to have added test coverage of these lines, since you are modifying them.